### ISSUE TITLE
TSX files do not report TypeScript diagnostics, while TS files work correctly (autocomplete still works)

### ISSUE THREAD

**Issue by pedroGoffi** [microsoft/typescript-go#4587]
Type: <b>Bug</b>

Install the TypeScript 7 (Native Preview) extension.
Open a React/Next.js TypeScript project.
Enable tsgo:
{
  "js/ts.experimental.useTsgo": true
}
Create a test.ts file:
const x: string = 123; // linter working completions OK

export default function Test() {
  const x: string = 123;

  return <div>{x}</div>;
}
Observe that:
Autocomplete, hover, go-to-definition, and semantic highlighting work correctly.
No TypeScript diagnostic is reported in the editor or the Problems panel for the type error in the .tsx file.
Check the TypeScript 7 output log. The server handles textDocument/diagnostic requests successfully and reports no internal errors.

Expected behavior

TypeScript diagnostics should be reported for .tsx files in the same way they are for .ts files.

Actual behavior

Diagnostics are reported for .ts files but not for .tsx files. Language features such as autocomplete and navigation continue to work normally.

Environment

VS Code: 1.128.0
TypeScript: 5.9.3
TypeScript 7 (Native Preview): 0.20260708.2
OS: Windows 11 x64
JSX: react-jsx
Project type: Next.js + React

Extension version: 0.20260708.2
VS Code version: Code 1.128.0 (fc3def6774c76082adf699d366f31a557ce5573f, 2026-07-07T15:14:24-07:00)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5500                                (12 x 3593)|
|GPU Status|2d_canvas: enabled<br>GPU0: VENDOR= 0x1002, DEVICE=0x6fdf [AMD Radeon RX 580 2048SP], DRIVER_VENDOR=AMD, DRIVER_VERSION=31.0.21925.1001 *ACTIVE*<br>GPU1: VENDOR= 0x1414, DEVICE=0x008c [Microsoft Basic Render Driver], DRIVER_VERSION=10.0.22621.5415<br>Machine model name: <br>Machine model version: <br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.89GB (4.44GB free)|
|Process Argv|. --crash-reporter-id f81a87e2-f303-4460-86b6-ebd65fd7684b|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythonrdcb7:31342333
6518g693:31463988
aj953862:31281341
4f60g487:31327383
6abeh943:31336334
envsactivate1:31538736
cloudbuttont:31379625
42190218_ostrepl:31403339
bj468946:31457056
ec5jj548:31422691
cp_cls_t_966_ss:31526232
4je02754:31466945
c3h7c220:31478652
38bie571_auto:31426784
cp_cls_c_1081:31454833
conptydll_true:31498968
e9c30283:31461165
test_treatment2:31471001
46204921:31447328
idci7584:31464702
e3e4d672:31494082
ei9d7968:31496641
chat:31457767
7e187181:31503455
ghj88844:31499326
h08i8180_copy:31475368
23c7c724:31520045
ddid_c:31478207
hmra_i5g22:31518061
7df3h592:31512476
cp_cls_t_1082:31535311
logging_enabled_new:31498466
j0d79568:31499440
jb_cp_cls_t_632:31543129
b1ei0813:31506238
748c7209:31512887
32d76977:31512328
ha629193:31508444
cdk-lw-on:31524445
f9bf7985:31543584
a1ije391_t:31540920
8hfba999:31537718
jbcp_cls_pctr_t:31531130
61138546:31518536
cp_intellij_t_nes:31548657
jf4hg949:31526829
h561g132:31533256
session_8:31532895
ihg5j128:31534457
f412h606:31546039
9gg7f176:31542112
nes-nc-ctrl:31546581

```

</details>

**Comment by weswigham** [comment]
AFAIK this is working as intended, assuming the `test.ts` file is excluded from your main react project's build, and you don't have another file in the react project open, the editor's active project set for the `Problems` pane will just be the inferred project with just the `.ts` file in it... And that's not a new behavior - that's true of TS6, too. Unless I'm missing something here? The OP is light on details for an actual standalone reproduction.

**Comment by elramus** [comment]
~~I can reproduce this with the following:~~

Update: Nevermind, false alarm 😅. It was an old/bad setting in my user settings JSON file. Sorry for the extra noise here. 

For future travelers just in case it's useful, I had `"javascript.validate.enable": false` because I didn't want JS files specifically to be checked. Looks like with the native preview that setting is now interpreted as a blanket disabling of typescript checking across the board.

**Comment by pedroGoffi** [comment]
used the same tsconfig for both everything was inside project include, but the linting was working only in the .ts files tho
